.github/workflows/ci.yml

```
@@ -76,6 +76,37 @@ jobs:
         run: |
           find "$GITHUB_WORKSPACE" -name "lcov.info"
 
+  build_keycloak_provider:
+    runs-on: ubuntu-latest
+    container:
+      image: maven:3.9.9-eclipse-temurin-21
+      options: --user root
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'adopt'
+          java-version: '21'
+
+      - name: Build with Maven
+        run: |
+          cd $GITHUB_WORKSPACE/resources/keycloak/custom-user-provider
+          mvn clean install
+
+      - name: Run Tests with Maven
+        run: |
+          cd $GITHUB_WORKSPACE/resources/keycloak/custom-user-provider
+          mvn test

```

chore: `mvn clean install` already runs the `test` task by default, so the `Run Tests with Maven` will cause the tests to run twice
```suggestion
      - name: Build and Test with Maven
        run: |
          cd $GITHUB_WORKSPACE/resources/keycloak/custom-user-provider
          mvn clean install
```

---

Updated!